ngraph/core/src/op/einsum.cpp

```diff
@@ -0,0 +1,299 @@
+
+    constructor_validate_and_infer_types();
+}
+
+bool op::v7::Einsum::is_subscript_correct(const std::string& subscript, bool& is_ellipsis_met)
+{
+    is_ellipsis_met = false;
+    auto subscript_length = subscript.length();
+    for (size_t ch_idx = 0; ch_idx < subscript_length; ++ch_idx)
+    {
+        if (is_ellipsis_met == false && ((subscript_length - ch_idx) > 2) &&
+            (subscript.substr(ch_idx, 3).compare("...") == 0))
+        {
+            // mark that ellipsis is met once
+            is_ellipsis_met = true;
+
+            // make additional increment since ellipsis consists of three dots.
+            ch_idx += 2;
+        }
+        else if (std::isalpha(subscript[ch_idx]) == 0)
+        {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+void op::v7::Einsum::parse_equation(const std::string& equation,
+                                    std::vector<std::string>& input_subscripts,
+                                    std::string& output_subscript)
+{
+    NGRAPH_OP_SCOPE(v7_Einsum_parse_equation);
+
+    // split equation to input subscripts and an output subscript
+    auto pos_output_delimeter = equation.find("->");
+    auto input_subscripts_str = equation.substr(0, pos_output_delimeter);
+
+    // split the input subscripts into a vector of input subscripts
+    bool is_ellipsis_met = false;
+    input_subscripts.clear();
+    std::istringstream input;
+    input.str(input_subscripts_str);
+    for (std::string input_subscript; std::getline(input, input_subscript, ',');)
+    {
+        bool local_is_ellipsis_met = false;
+        // check that input subscript contains only alphabetic letter or ellipsis
+        NGRAPH_CHECK(is_subscript_correct(input_subscript, local_is_ellipsis_met) == true,
+                     "Input subscript of Einsum equation must consist of either only "
+                     "alphabetic letters or alphabetic letters with one ellipsis.");
+
+        // mark that ellipsis is met at least in one input subscript
+        if (local_is_ellipsis_met == true)
+        {
+            is_ellipsis_met = true;
+        }
+        input_subscripts.push_back(input_subscript);
+    }
+
+    if (pos_output_delimeter == std::string::npos)
+    {
+        // recover output subscript
+        output_subscript = "";
+        for (auto const& input_subscript : input_subscripts)
+        {
+            for (auto const& label : input_subscript)
+            {
+                if (std::isalpha(label) && output_subscript.find(label) == std::string::npos)
+                {
+                    output_subscript += label;
+                }
+            }
+        }
+        std::sort(output_subscript.begin(), output_subscript.end());
+        if (is_ellipsis_met)
+        {
+            output_subscript = "..." + output_subscript;
+        }
+    }
+    else
+    {
+        output_subscript = equation.substr(pos_output_delimeter + 2);
+        bool local_is_ellipsis_met = false;
```

Comment:
may be call this variable `output_subscript_ellipsis_met`?

Reply:
from the function description called for this variable it is evident that this is done for output subscript.